agent/component/base.py

```diff
@@ -416,7 +418,7 @@ def run(self, history, **kwargs):
     def _run(self, history, **kwargs):
         raise NotImplementedError()
 
-    def output(self, allow_partial=True) -> tuple[str, pd.DataFrame | partial]:
+    def output(self, allow_partial=True) -> Tuple[str, Union[pd.DataFrame, partial]]:
```

Comment:
Use `tuple` instead of `Tuple`.
Use `pd.DataFrame | partial` instead of `Union[pd.DataFrame, partial]]`.